week9/d01/Assignment_9_1_1/ex2.cpp

```diff
@@ -0,0 +1,39 @@
+#include "includes/math.hpp"
+#include <iostream>
+#include <unordered_map>
+#include <functional>
+#include <type_traits>
+#include <limits>
+
+int main(int argc, char* argv[])
+{
+    std::unordered_map<char, std::function<int(int,int)>> functions{
+        {'+', sum}, {'-', substract}, {'*', multiply}, {'/', divide}};
+    
+    if (argc != 4)
+    {
+        std::cout << "This program takes 3 arguments!\n";
+        std::cout << "For example:\n";
+        std::cout << "./calculator 5 \\* 5\n";
+        return 1;
+    }
+    std::string arg1 = argv[1];
+    std::string arg3 = argv[3];
+    if (!is_int(arg1) || !is_int(arg3))
+    {
+        std::cout << "First and third parameters have to be integers!\n";
+        return 1;
+    }
+    const char* arg2 = argv[2];
+    if (*arg2 == '+' || *arg2 == '-' || *arg2 == '*' || *arg2 == '/')
```

Comment:
Should probably check just if the map has a key `*arg2`, instead of hand-writing every option.  Also gives better opportunity to add new functionality (since you just need to update the map)